.github/workflows/build.yml

```diff
@@ -1,42 +1,121 @@
+---
 name: Build
 
 on:
-  push:
-    branches: [ '*' ]
   pull_request:
```

Comment:
I think this should probably only run on push to main because we need to ensure we time updating the docs with when we release

Reply:
This is building when we create a pull request. The publish will happen in another job `publish` only on master